test/common/common/log_verbosity_update_test.cc

```diff
@@ -0,0 +1,229 @@
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+
+  std::pair<absl::string_view, int> update_5 = std::make_pair("/*", 1);
+  getFancyContext().updateVerbositySetting({update_5});
+
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+}
+
+TEST(Fancy, globQuestionUpdate) {
+  const char* file_1 = "envoy/src/foo.cc";
+  const char* file_2 = "envoy/f__.cc";
+  const char* file_3 = "/bar.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+  getFancyContext().initFancyLogger(file_3, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("f??", 3);
+  getFancyContext().updateVerbositySetting({update_1});
+
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+
+  std::pair<absl::string_view, int> update_2 = std::make_pair("????y/*", 2);
+  getFancyContext().updateVerbositySetting({update_2});
+  p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_EQ(p->level(), spdlog::level::info);
+  p = getFancyContext().getFancyLogEntry(file_3);
+  EXPECT_EQ(p->level(), Logger::Context::getFancyDefaultLevel());
+}
+
+TEST(Fancy, inOrderUpdate) {
+  const char* file_1 = "envoy/src/foo.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("f??", 3);
+  std::pair<absl::string_view, int> update_2 = std::make_pair("envoy/*/f??", 2);
+  getFancyContext().updateVerbositySetting({update_1, update_2});
+
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+}
+
+TEST(Fancy, invalidLevelUpdate) {
+  const char* file_1 = "envoy/src/foo.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("f??", 9);
```

Comment:
const (here and elsewhere if not mutating the value)

Reply:
gotcha